app/pb/threads.proto

```diff
@@ -25,7 +25,9 @@ service Threads {
 
   // The same API is used regardless of level. All posts are using the same id namespace.
 
-  // Get non-deleted posts belonging to a thread, sorted in reverse time order.
+  // Get non-deleted posts belonging to a thread.
+  // For ease of display, the pagination order is most recent first, but within a page, replies are oldest first.
+  // For example, if A is the most recent reply, the first page would have `[C, B, A]` and the second page `[F, E, D]`.
```

Comment:
Crap, so with the paginated query on the frontend, that means we're gonna get [C, B, A, F, E, D] when we do combine the lists before we render it on the page. But that's not gonna be correct right as we'd want [F, E, D, C, B, A]?

To get the correct thing, we'd have to go through those pages backwards but since you can't write for-loops in JSX, that means the frontend would still need to reverse the pages first before this will be right... 🤔

Reply:
True. Maybe reversing in the frontend is not so bad?
I don't think we're likely to have over 1000 replies on screen. 

Maybe we can do some kind of custom iterator later if there are performance issues?

Reply:
Yeah thinking about this more, I think doing things on the backend will only work if it's returning data consistently in the same direction as we go through pages, but I don't think that's possible without breaking the pagination logic in this case, as you'd have to ascending order and somehow know to return the last X items, which would do a full table scan and would kill performance faster than the frontend lol so that's not really an option.

So yeah maybe it is easier to just leave it and do it on the frontend. I reckon we will run into more performance issues with 1000s of replies on the screen itself (i.e. in the DOM) rather than a couple of linear time list operations tbh, but that's a different problem to solve for us (and a good problem to have in the famous words of Aapeli 😂 )